server/src/services/listings.ts

```diff
@@ -14,16 +14,23 @@
  * limitations under the License.
  */
 
+import schedule from 'node-schedule';
+
 import {DEFAULT_LISTING_PAYLOAD} from '../constants/default-payload';
 import {Filter, ListingRequest, ListingResponse} from '../interfaces';
-import {listingStorage} from '../storage';
+import {commitStorage, listingStorage} from '../storage';
 
-const addListing = async (listing: ListingRequest): Promise<ListingResponse> =>
-  listingStorage.addListing({
+const addListing = async (
+  listing: ListingRequest
+): Promise<ListingResponse> => {
+  const addedListing = await listingStorage.addListing({
     ...listing,
     ...DEFAULT_LISTING_PAYLOAD,
   });
-
+  const {id, deadline} = addedListing;
+  schedule.scheduleJob(deadline, () => updateListingSuccess(id));
```

Comment:
how does this work?
will it be persisted and resumed if server stopped and restarted?

Reply:
Same question.

Reply:
@ifhuang @xlai20 It won't be persisted. The last time @hzxcaryn and I discussed about this, we thought that we can have a script to reschedule the jobs if the server stopped and restarted.

Alternatively, we can also have cron job, but they will have to run every day, whether or not there is any listing to update and it will also have one additional step of filtering listings which are past deadline and thus need to be updated.

Reply:
But I think we previously didn't consider that GAE might create and shut down instances to manage traffic. In that case, I don't think our solution works. I guess we might still need to have a cron job after all.